From 2e1746da18581160d129f6264dac14245350b672 Mon Sep 17 00:00:00 2001
From: baudurie <benjamin.audurier@cern.ch>
Date: Wed, 15 Jun 2016 14:32:40 +0200
Subject: [PATCH 17/36] small fix in MUL selection event cut

---
 PWG/muon/AliAnalysisMuMuEventCutter.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/PWG/muon/AliAnalysisMuMuEventCutter.cxx b/PWG/muon/AliAnalysisMuMuEventCutter.cxx
index 1e618de..e934674 100644
--- a/PWG/muon/AliAnalysisMuMuEventCutter.cxx
+++ b/PWG/muon/AliAnalysisMuMuEventCutter.cxx
@@ -194,7 +194,7 @@ Bool_t AliAnalysisMuMuEventCutter::IsPhysicsSelectedINT8(const AliInputEventHand
 Bool_t AliAnalysisMuMuEventCutter::IsPhysicsSelectedMUL(const AliInputEventHandler& eventHandler) const
 {
   /// Whether or not the event is physics selected
-  return const_cast<AliInputEventHandler&>(eventHandler).IsEventSelected() & AliVEvent::kMuonUnlikePB;
+  return const_cast<AliInputEventHandler&>(eventHandler).IsEventSelected() & AliVEvent::kMuonUnlikeLowPt7;
 }
 
 Bool_t AliAnalysisMuMuEventCutter::IsPhysicsSelectedMUSPB(const AliInputEventHandler& eventHandler) const
-- 
2.7.4 (Apple Git-66)

