From 1a38990668289567415f2b88032a4d45f041900f Mon Sep 17 00:00:00 2001
From: baudurie <benjamin.audurier@cern.ch>
Date: Sat, 18 Jun 2016 17:52:14 +0200
Subject: [PATCH 30/36] Small fix

---
 PWG/muondep/AliAnalysisMuMuSpectraCapsulePP.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/PWG/muondep/AliAnalysisMuMuSpectraCapsulePP.cxx b/PWG/muondep/AliAnalysisMuMuSpectraCapsulePP.cxx
index 9a12aed..39b6b03 100644
--- a/PWG/muondep/AliAnalysisMuMuSpectraCapsulePP.cxx
+++ b/PWG/muondep/AliAnalysisMuMuSpectraCapsulePP.cxx
@@ -160,7 +160,7 @@ TList* AliAnalysisMuMuSpectraCapsulePP::ComputePPCrossSection(const char* what)
         if(fPrintFlag){
           printf("\n");
           printf("%s                        = %f +/- %f\n",what,r->GetValue(what),r->GetErrorStat(what));
-          printf("NofJPsi                   = %f +/- %f +/- %f \n ",NofJPsi,NofJPsiError,r->GetRMS("NofJPsi"));
+          printf("NofJPsi                   = %f +/- %f (%f %%)+/- %f (%f %%)\n ",NofJPsi,NofJPsiError,100*NofJPsiError/NofJPsi,r->GetRMS("NofJPsi"),100*r->GetRMS("NofJPsi")/NofJPsi);
           printf("Systematic MC             = %f \n ",valueArray[0]);
           printf("Tracking Error            = %f \n ",valueArray[1]);
           printf("Trigger  Error            = %f \n ",valueArray[2]);
@@ -200,7 +200,7 @@ TList* AliAnalysisMuMuSpectraCapsulePP::ComputePPCrossSection(const char* what)
         // Uncorrelated error squared (%)
         Double_t UncError2 =
           (r->GetRMS("NofJPsi")/r->GetValue("NofJPsi"))*(r->GetRMS("NofJPsi")/r->GetValue("NofJPsi")) // Signal 
-          + pow(valueArray[0]/100.,2)  // systemtic MC(%) 
+          + pow(valueArray[0]/100.,2)  // systematic MC(%) 
           + pow(valueArray[1]/100.,2)  // Tracking Error(%)
           + pow(valueArray[2]/100.,2)  // Trigger  Error(%) (trigger response)
           + pow(valueArray[3]/100.,2)  // matching Error(%)
-- 
2.7.4 (Apple Git-66)

